### DISCUSSION TITLE
User-defined combinations of other combinations

### DISCUSSION THREAD

**Discussion by NextFEM** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#236] [General]
Actually, you can define envelopes, but only via load cases + national standard. That means, that you define load cases you want to use and definition how to create the specific combinations are defined by national code.

To allow also software that has not been implemented on the basis of national standards, it would be great to have user-defined combination of other combinations, as to let the user to define also custom envelopes.

**Comment by robinblazek** [comment]
Hi @NextFEM,
if I understand it correctly, it seems to me like "Result class" object, that is already proposed here: [Result class](https://github.com/StructuralAnalysisFormat/StructuralAnalysisFormat-Doc/discussions/235)
Please, check its definition, if it sufficiently meets your requirements or not and let us know. 
Thanks for your feedback!

**Comment by NextFEM** [comment]
Hello @robinblazek ,
I read this, and this is part of what I mean. The main focus in my proposal is to un-tie the format from national standards, it means for envelopes set and also for combinations set. Also to improve readability to other countries.
If you believe this is a duplicate please close. Result class is a slightly misleading title, but I completely agree on its structure, except for a missing property containing coefficient, that is needed.

**Comment by MartinSalak** [comment]
Hello @NextFEM,
Thank you for your feedback. It is very valuable to us. Could you please help us understand what exactly are you missing in SAF? Let me elaborate on one example:

Envelopes without national standards:
I have these 6 load cases:
![image](https://user-images.githubusercontent.com/89633938/203931237-429c24b2-d59d-4d01-81c8-06455e71c96a.png)
they are in 3 load groups
![image](https://user-images.githubusercontent.com/89633938/203932900-9028485e-b6f0-4810-8e92-fb222cd7e9eb.png)
now I put all the load cases into one **envelope** combination together with the correct coefficient for each load case
![image](https://user-images.githubusercontent.com/89633938/203933334-840b6484-6357-4e1f-9916-12834abc358b.png)
the row continues to the right and contains this:
![image](https://user-images.githubusercontent.com/89633938/203933618-3518f15c-fd20-4f02-8b53-9e9fcb1925d1.png)

An envelope combination is a combination of linear combinations. The rules of creating the combinations are set in load groups: LC1 and LC2 are always together and LC5 and LC6 are never together. In this envelope, there will be a combination of many linear combinations. For example <LC1+LC2> <LC1+LC2+LC3> <LC1+LC2+LC4> <LC1+LC2+LC3+LC4+LC5> and so on. The coefficients are assigned to each load case using the properties Load factor X and Multiplier X.

Is this not the result you described? Is there anything more you would need in SAF? Does NextFEM work like this with combinations? Would you prefer to have the possibility to create a set of these envelopes and assign coefficient to the whole envelope combination? If yes, why?

**Reply by NextFEM** [reply]
Hello,
thanks for the reply. It could be suitable in the way you describe, of course. 
In general, we allow user to set a coefficient directly in the envelope combination, for each linear combinations involved. This covers particulars need (e.g. scaling of maximum forces on a member), that for sure are not related to combinations given by codes.
Without introducing this coefficient, in our case it is a lost information (since it is assumed as 1) but for the vast majority of cases it could fit the needs.
